### ISSUE TITLE
Issue with finally block and no-empty

### ISSUE THREAD

**Issue by zakangelle** [babel/babel-eslint#26]
If `no-empty` is set to `1` (default), a `finally` block will cause the error below. Here is a [small test](https://github.com/zakangelle/babel-eslint-finally-issue) to reproduce it.

test.js:
```js
function test() {};

try {
  test();
}
catch(e) {
}
finally {
}
```

```
.../lib/node_modules/eslint/lib/rules/no-empty.js:23
                    (isFinallyBlock && !parent.handlers.length)) {
                                                       ^
TypeError: Cannot read property 'length' of undefined
    at EventEmitter.BlockStatement (.../lib/node_modules/eslint/lib/rules/no-empty.js:23:56)
    at EventEmitter.emit (events.js:129:20)
    at Controller.controller.traverse.enter (.../lib/node_modules/eslint/lib/eslint.js:683:25)
    at Controller.__execute (.../lib/node_modules/eslint/node_modules/estraverse/estraverse.js:397:31)
    at Controller.traverse (.../lib/node_modules/eslint/node_modules/estraverse/estraverse.js:495:28)
    at EventEmitter.module.exports.api.verify (.../lib/node_modules/eslint/lib/eslint.js:676:24)
    at processFile (../lib/node_modules/eslint/lib/cli-engine.js:172:23)
    at ..lib/node_modules/eslint/lib/cli-engine.js:271:26
    at walk (../lib/node_modules/eslint/lib/util/traverse.js:81:9)
    at .../lib/node_modules/eslint/lib/util/traverse.js:102:9
```

**Comment by ivanoats** [comment]
I can confirm this issue, and that setting no-empty to 0 is a temporary work-around.

**Comment by sebmck** [comment]
Fixed via eslint/eslint#1930.
